src/platforms/stm32/src/lib/gpio_driver.c

```diff
@@ -721,59 +759,59 @@ static term gpiodriver_set_int(Context *ctx, int32_t target_pid, term cmd)
     term gpio_bank_atom = term_get_tuple_element(gpio_tuple, 0);
     if (UNLIKELY(!term_is_atom(gpio_bank_atom))) {
         AVM_LOGE(TAG, "Bank parameter of pin tuple must be an atom!");
-        return create_pair(ctx, ERROR_ATOM, INVALID_BANK_ATOM);
+        return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_bank_atom));
     }
     uint32_t gpio_bank = (uint32_t) interop_atom_term_select_int(gpio_bank_table, gpio_bank_atom, ctx->global);
     if (UNLIKELY(gpio_bank == GPIOInvalidBank)) {
         char *bank_string = interop_atom_to_string(ctx, gpio_bank_atom);
         AVM_LOGE(TAG, "Invalid GPIO bank '%s' in pin tuple", bank_string);
         free(bank_string);
-        return create_pair(ctx, ERROR_ATOM, INVALID_BANK_ATOM);
+        return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_bank_atom));
     }
 
     uint16_t gpio_pin = (uint16_t) term_to_int32(term_get_tuple_element(gpio_tuple, 1));
 
     struct GPIOData *gpio_data = ctx->platform_data;
     if (gpiodriver_is_gpio_attached(gpio_data, gpio_bank_atom, gpio_pin)) {
         char *bank_string = interop_atom_to_string(ctx, gpio_bank_atom);
-        AVM_LOGW(TAG, "GPIO pin %c%u interrupt already attached.", bank_string, gpio_pin);
+        AVM_LOGW(TAG, "GPIO pin %s%u interrupt already attached.", bank_string, gpio_pin);
         free(bank_string);
         return OK_ATOM;
     }
 
     term trigger = term_get_tuple_element(cmd, 2);
     if (UNLIKELY(!term_is_atom(trigger))) {
         AVM_LOGE(TAG, "GPIO interrupt trigger must be an atom ('rising', 'falling', or 'both').");
-        return create_pair(ctx, ERROR_ATOM, INVALID_TRIGGER_ATOM);
+        return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_trigger_atom));
     }
     enum exti_trigger_type interrupt_type = interop_atom_term_select_int(exti_trigger_table, trigger, ctx->global);
     if (UNLIKELY(interrupt_type == INVALID_EXTI_TRIGGER)) {
         char *trigger_string = interop_atom_to_string(ctx, trigger);
         AVM_LOGE(TAG, "Interrupt type %s not supported on stm32 platform.", trigger_string);
         free(trigger_string);
-        return create_pair(ctx, ERROR_ATOM, INVALID_TRIGGER_ATOM);
+        return create_pair(ctx, ERROR_ATOM, globalcontext_make_atom(ctx->global, invalid_trigger_atom));
     }
 
     uint32_t exti = 1U << gpio_pin;
 
     if (!list_is_empty(&gpio_data->gpio_listeners)) {
         struct ListHead *item;
         struct ListHead *tmp;
```

Comment:
You can probably get rid of the declaration of `tmp`.  Does this yield a compiler warning?  (Not that you could tell on STM32, right?  :)

Reply:
Oops! I just missed that unneeded `tmp`, I'm sure that did add a warning that I just missed in the wall of warning.

Reply:
fixed.